### PR TITLE
Generate Nullable annotation at constructor arguments.

### PR DESCRIPTION
**Pull request by mik9**
Generate @Nullable annotation at constructor arguments.

AutoValue itself does this for generated class.
Also when we use @ParametersAreNonnullByDefault in project then IDE and Spotbugs will complain when we try to pass null as constructor argument which not annotated with @Nullable (but appropriate abstract method does annotated).

### CODE REVIEW

**Comment by hzsweers** [comment]
Could you add a description?

**Comment by mik9** [comment]
Updated description.

File: auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java [hunk 1]
```diff
@@ -157,15 +157,23 @@ public boolean shouldDeserialize() {
...
+      return !nullableAnnotation().isEmpty();
+    }
+
+    public String nullableAnnotation() {
+      for (String annotationString : annotations) {
+        if (annotationString.equals("@Nullable") || annotationString.endsWith(".Nullable")) {
+          return annotationString;
+        }
+      }
+      return "";
```
Comment (hzsweers) [inline auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: This should return null if not found, not an empty string

File: auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java [hunk 2]
```diff
@@ -157,15 +157,23 @@ public boolean shouldDeserialize() {
     }
 
     public boolean nullable() {
-      return annotations.contains("Nullable");
+      return !nullableAnnotation().isEmpty();
```
Comment (hzsweers) [inline auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Can we pre-calculate this and save it?

Comment (mik9) [inline auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java]: Sure, I just took code from AutoValue's Property class.

**Comment by mik9** [comment]
Hello, any more changes I need to make to allow this to be merged?

**Comment by hzsweers** [comment]
Sorry I didn't see you'd updated it

**Comment by hzsweers** [comment]
Thanks!

**Comment by mik9** [comment]
Np, thanks! And same PR for AutoValue Gson: https://github.com/rharter/auto-value-parcel/pull/124

**Pull request merged**